include/dlaf/eigensolver/band_to_tridiag/mc.h

```diff
@@ -365,6 +531,101 @@ class SweepWorker {
   memory::MemoryView<T, Device::CPU> data_;
 };
 
+template <class T>
+class SweepWorkerDist : private SweepWorker<T> {
+public:
+  SweepWorkerDist(SizeType size, SizeType band_size) : SweepWorker<T>(size, band_size) {}
+
+  void startSweep(SizeType sweep, BandBlockDist<T>& a) {
+    this->startSweepInternal(sweep, a);
+  }
+
+  void doStep(BandBlockDist<T>& a) noexcept {
+    SizeType j = this->firstRowHHR();
+    SizeType n = this->sizeHHR();  // size diagonal tile and width off-diag tile
+
+    const auto n1 = a.nextSplit(j);
+    if (n1 < n) {
+      doStepSplit(a, n1);
+    }
+    else {
+      this->doStepFull(a);
+    }
+  }
+
+  void send(const comm::Communicator& comm, comm::IndexT_MPI dest, comm::IndexT_MPI tag,
+            MPI_Request* req) const noexcept {
+    DLAF_MPI_CHECK_ERROR(MPI_Isend(const_cast<T*>(data_()), to_int(band_size_ + 1),
```

Comment:
How sure are we that this `const_cast` can't be used wrongly? Why not
```suggestion
            MPI_Request* req) noexcept {
    DLAF_MPI_CHECK_ERROR(MPI_Isend(data_(), to_int(band_size_ + 1),
```
?

Reply:
well spotted... the const cast should not be there (don't remember why I put it there during debugging)... the first parameter of Isend is a `const void*`.

Reply:
Fixing it in `include/dlaf/communication/kernels/p2p.h` as well..